plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/nessie/TestIcebergNessieCatalogWithAuth.java

```diff
@@ -0,0 +1,140 @@
+                .put("QUARKUS_OIDC_CLIENT_ID", "projectnessie")
+                .put("NESSIE_SERVER_AUTHENTICATION_ENABLED", "true")
+                .buildOrThrow();
+
+        NessieContainer nessieContainer =
+                closeAfterClass(NessieContainer.builder().withEnvVars(envVars).withNetwork(network).build());
+        nessieContainer.start();
+        nessieUri = nessieContainer.getRestApiUri();
+
+        tempDir = Files.createTempDirectory("test_trino_nessie_catalog");
+
+        return getDistributedQueryRunner(nessieUri, true, keycloakContainer.getAccessToken());
+    }
+
+    private DistributedQueryRunner getDistributedQueryRunner(String nessieUri, boolean isBearerAuth, String accessToken)
+            throws Exception
+    {
+        ImmutableMap.Builder<String, String> properties = ImmutableMap.builder();
+        properties.put("iceberg.file-format", "PARQUET")
+                .put("iceberg.catalog.type", "nessie")
+                .put("iceberg.nessie-catalog.uri", nessieUri)
+                .put("iceberg.nessie-catalog.default-warehouse-dir", tempDir.toString())
+                .put("iceberg.writer-sort-buffer-size", "1MB");
+        if (isBearerAuth) {
+            properties.put("iceberg.nessie-catalog.authentication.type", "BEARER");
+        }
+        if (accessToken != null) {
+            properties.put("iceberg.nessie-catalog.authentication.token", accessToken);
+        }
+
+        return IcebergQueryRunner.builder()
+                .setBaseDataDir(Optional.of(tempDir))
+                .setIcebergProperties(properties.buildOrThrow())
+                .build();
+    }
+
+    @Test
+    public void testWithInvalidAccessToken()
+    {
+        assertThatThrownBy(() -> getDistributedQueryRunner(nessieUri, true, "invalid_token"))
+                .isInstanceOf(QueryFailedException.class)
+                .hasMessageContaining("Unauthorized (HTTP/401)");
+    }
+
+    @Test
+    public void testWithNoAccessToken()
+    {
+        assertThatThrownBy(() -> getDistributedQueryRunner(nessieUri, true, null))
+                .isInstanceOf(CreationException.class)
+                .hasMessageContaining("'iceberg.nessie-catalog.authentication.token' must be configured with 'iceberg.nessie-catalog.authentication.type' BEARER");
+    }
+
+    @Test
+    public void testWithNoAuthAndAccessToken()
+    {
+        assertThatThrownBy(() -> getDistributedQueryRunner(nessieUri, false, "some_token"))
+                .isInstanceOf(CreationException.class)
+                .hasMessageContaining("'iceberg.nessie-catalog.authentication.token' must be configured only with 'iceberg.nessie-catalog.authentication.type' BEARER");
+    }
+
+    @Test
+    public void testWithValidAccessToken()
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "hidden_file_path", "(a int, b VARCHAR)", ImmutableList.of("(1, 'a')"))) {
```

Comment:
I'm not sure why hidden file path is used for verifying access token. Just using simple SELECT query looks fine in my opinion.